src/handlers/meta_transaction_handlers.ts

```diff
@@ -237,3 +373,129 @@ function parseV1RequestParams(req: express.Request): MetaTransactionV1QuoteReque
+                        reason: ValidationErrorReasons.InvalidGaslessFeeKind,
+                    },
+                ]);
+            }
+
+            // ASK: 0x-api has been using 0-1 for percentage instead of 0-100. Should we use
+            //      0-1 here to be consistent with other fields?
+            const volumePercentage = new BigNumber(integratorFee.volumePercentage as string);
+            if (volumePercentage.gt(1)) {
+                throw new ValidationError([
+                    {
+                        field: 'feeConfigs',
+                        code: ValidationErrorCodes.ValueOutOfRange,
+                        reason: ValidationErrorReasons.PercentageOutOfRange,
+                    },
+                ]);
+            }
+
+            parsedFeeConfigs.integrator = {
+                kind: 'volume',
+                feeToken: integratorFee.feeToken as string,
+                feeRecipient: integratorFee.feeRecipient as string,
+                volumePercentage,
+            };
+        }
+
+        // Parse the 0x fee config
+        if (feeConfigs.zeroex) {
+            const zeroexFee: ParsedQs = feeConfigs.zeroex as ParsedQs;
+
+            if (zeroexFee.kind !== 'volume' && zeroexFee.kind !== 'integrator_share') {
+                throw new ValidationError([
+                    {
+                        field: 'feeConfigs',
+                        code: ValidationErrorCodes.IncorrectFormat,
+                        reason: ValidationErrorReasons.InvalidGaslessFeeKind,
+                    },
+                ]);
+            }
+
+            const feePercentage = new BigNumber(zeroexFee.volumePercentage as string);
+            if (feePercentage.gt(1)) {
+                throw new ValidationError([
+                    {
+                        field: 'feeConfigs',
+                        code: ValidationErrorCodes.ValueOutOfRange,
+                        reason: ValidationErrorReasons.PercentageOutOfRange,
+                    },
+                ]);
+            }
+
+            if (zeroexFee.kind === 'volume') {
+                parsedFeeConfigs.zeroex = {
+                    kind: 'volume',
+                    feeToken: zeroexFee.feeToken as string,
+                    feeRecipient: zeroexFee.feeRecipient as string,
+                    volumePercentage: feePercentage,
+                };
+            } else if (zeroexFee.kind === 'integrator_share') {
+                parsedFeeConfigs.zeroex = {
+                    kind: 'integrator_share',
+                    feeToken: zeroexFee.feeToken as string,
+                    feeRecipient: zeroexFee.feeRecipient as string,
+                    integratorSharePercentage: new BigNumber(zeroexFee.integratorSharePercentage as string),
+                };
+            }
+        }
+
+        // Parse the gas fee config
+        if (feeConfigs.gas) {
+            const gasFee: ParsedQs = feeConfigs.gas as ParsedQs;
+
+            if (gasFee.kind !== 'gas') {
```

Comment:
can gasFee ever not be gas?

Reply:
If no mistake is made, then `gasFee` should always have kind `gas`. Again trying to be extra safe here in the case of mistake and surface the error to the caller